internal/handlers/pg/pg.go

```diff
@@ -41,13 +48,45 @@ type NewOpts struct {
 func New(opts *NewOpts) (handlers.Interface, error) {
 	h := &Handler{
 		NewOpts: opts,
+		pools:   make(map[string]*pgdb.Pool, 1),
 	}
 	return h, nil
 }
 
 // Close implements HandlerInterface.
 func (h *Handler) Close() {
-	h.PgPool.Close()
+	h.rw.Lock()
+	defer h.rw.Unlock()
+
+	for k, pgPool := range h.pools {
+		pgPool.Close()
+		delete(h.pools, k)
+	}
+}
+
+// DBPool returns database connection pool for the given client connection.
+func (h *Handler) DBPool(ctx context.Context) (*pgdb.Pool, error) {
+	// TODO make real implementation; the current one is a stub
```

Comment:
I understand that this is not for the current PR, as the current PR is a stub.
As no issue is mentioned, I expect it to be fixed in the next PR (that we start passing auth data from the client).

Reply:
#1725